js/app/environment.js

```diff
@@ -198,4 +199,17 @@ environment.render = function () {
   })
 }
 
+environment.markDreamAsViewed = function (dreamIndex) {
+  var clonedGeom = this.dreamsMesh.geometry.clone()
+  this.removeObjectFromScene(this.dreamsMesh)
+  var faceIndices = this.pickingData[dreamIndex].faceIndices
+  range(faceIndices.low, faceIndices.hi).forEach(function (faceIndex) {
+    clonedGeom.faces[faceIndex].materialIndex = 1
```

Comment:
Yeah, It is interesting how we have to do this. Like, I feel like we should be able to just update the material in the same way that you can change the opacity of a material in run time and it updates. I couldn't get it to work like that though. I feel like maybe this is a quirk of the multiple material biz. I though that just updating the geometry and then creating a new mesh with a new material and the old geom would work, but that doesn't register the change in material indexes. It must compile that information when the geometry is instantiated, or something of the like.